src/cargo-kani/src/args.rs

```diff
@@ -0,0 +1,313 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use anyhow::bail;
+use clap::{arg_enum, Error, ErrorKind};
+use std::ffi::OsString;
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+#[derive(Debug, StructOpt)]
+#[structopt(
+    name = "kani",
+    about = "Verify a single Rust file. For more information, see https://github.com/model-checking/kani"
+)]
+pub struct StandaloneArgs {
+    /// Rust file to verify
+    #[structopt(parse(from_os_str))]
+    pub input: PathBuf,
+
+    #[structopt(flatten)]
+    pub common_opts: KaniArgs,
+}
+
+#[derive(Debug, StructOpt)]
+#[structopt(
+    name = "cargo-kani",
+    about = "Verify a Rust crate. For more information, see https://github.com/model-checking/kani"
+)]
+pub struct CargoKaniArgs {
+    #[structopt(flatten)]
+    pub common_opts: KaniArgs,
+}
+
+// Common arguments for invoking Kani. This gets put into KaniContext, whereas
+// anything above is "local" to "main"'s control flow.
+#[derive(Debug, StructOpt)]
+pub struct KaniArgs {
```

Comment:
What is the output for `kani --help` after this change?

Reply:
Right now a big jumbled mess:

```
$ cargo-kani --help
cargo-kani 0.1.0
Verify a Rust crate. For more information, see https://github.com/model-checking/kani

USAGE:
    cargo-kani [FLAGS] [OPTIONS]

FLAGS:
        --allow-cbmc-verification-failure    Do not produce error return code on CBMC verification failure
        --assertion-reach-checks             Turn on assertion reachability checks
        --auto-unwind                        Turn on automatic loop unwinding
        --debug                              Produce full debug information
        --default-checks                     Turn on all default checks
        --dry-run                            Print commands instead of running them
        --gen-c                              Generate C file equivalent to inputted program
    -h, --help                               Prints help information
        --keep-temps                         Keep temporary files generated throughout Kani process
        --memory-safety-checks               Turn on default memory safety checks
        --no-default-checks                  Turn off all default checks
        --no-memory-safety-checks            Turn off default memory safety checks
        --no-overflow-checks                 Turn off default overflow checks
        --no-restrict-vtable                 Disable restricting the targets of virtual table function pointer calls
        --no-undefined-function-checks       Turn off undefined function checks
        --no-unwinding-checks                Turn off default unwinding checks
        --only-codegen                       Kani will only compile the crate
        --overflow-checks                    Turn on default overflow checks
    -q, --quiet                              Produces no output, just an exit code and requested artifacts; overrides
                                             --verbose
        --restrict-vtable                    Restrict the targets of virtual table function pointer calls
        --tests                              Enable test function verification. Only use this option when the entry
                                             point is a test function
        --undefined-function-checks          Turn on undefined function checks
        --unwinding-checks                   Turn on default unwinding checks
        --use-abs                            Use abstractions for the standard library
    -V, --version                            Prints version information
    -v, --verbose                            Output processing stages and commands, along with minor debug information
        --visualize                          Generate visualizer report to <target-dir>/report/html/index.html

OPTIONS:
        --abs-type <abs-type>              Choose abstraction for modules of standard library if available [default:
                                           std]  [possible values: std, kani, c-ffi, no-back]
        --c-lib <c-lib>...                 Link external C files referenced by Rust code
        --cbmc-args <cbmc-args>...         Pass through directly to CBMC; must be the last flag
        --function <function>              Entry point for verification [default: main]
        --object-bits <object-bits>        Specify the number of bits used for representing object IDs in CBMC [default:
                                           16]
        --output-format <output-format>    Toggle between different styles of output [default: regular]  [possible
                                           values: Regular, Terse, Old]
        --target-dir <target-dir>          Directory for all generated artifacts
        --unwind <unwind>                  Specify the value used for loop unwinding in CBMC
```

IMO, cleaning it up is another PR.